core/include/kernel/interrupt.h

```diff
@@ -258,6 +258,70 @@ static inline void interrupt_disable(struct itr_chip *chip, size_t itr_num)
 	chip->ops->disable(chip, itr_num);
 }
 
+/*
+ * interrupt_can_raise_pi() - Return whether controller embeds raise_pi
+ * @chip	Interrupt controller
+ */
+static inline bool interrupt_can_raise_pi(struct itr_chip *chip)
+{
+	return chip->ops->raise_pi;
+}
+
+/*
+ * interrupt_can_raise_sgi() - Return whether controller embeds raise_sgi
+ * @chip	Interrupt controller
+ */
+static inline bool interrupt_can_raise_sgi(struct itr_chip *chip)
+{
+	return chip->ops->raise_sgi;
+}
+
+/*
+ * interrupt_can_set_affinity() - Return whether controller embeds set_affinity
+ * @chip	Interrupt controller
+ */
+static inline bool interrupt_can_set_affinity(struct itr_chip *chip)
+{
+	return chip->ops->set_affinity;
+}
+
+/*
+ * interrupt_raise_pi() - Raise a peripheral interrupt of a controller
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number to raise
+ */
+static inline void interrupt_raise_pi(struct itr_chip *chip, size_t itr_num)
+{
+	assert(interrupt_can_raise_pi(chip));
+	chip->ops->raise_pi(chip, itr_num);
+}
+
+/*
+ * interrupt_raise_sgi() - Raise a software generiated interrupt of a controller
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number to raise
+ * @cpu_mask	Mask of the CPUs targeted by the interrupt
+ */
+static inline void interrupt_raise_sgi(struct itr_chip *chip, size_t itr_num,
+				       uint8_t cpu_mask)
```

Comment:
Not strictly related to this PR, but... I notice that `cpu_mask` is an `uint8_t`, does it mean the API can only support at most 8 cores? Or perhaps a "CPU" is not a core? (the comment near `itr_raise_sgi()` mentions a "cpu interface").

Reply:
Actually I mimicked the existing GIC function API. As for a generic purpose interrupt controller framework, I don't think "raise SGI" or "raise PPI" really make sense but I propose to postpone this to later, once there's a need.